manager/controlapi/secret.go

```diff
@@ -31,6 +31,34 @@ func secretFromSecretSpec(spec *api.SecretSpec) *api.Secret {
 	}
 }
 
+// checkSecretInUse does a best effort to find if the passed in secret is in
+// use by any services
+func (s *Server) checkSecretInUse(secretID string) error {
+	var (
+		services []*api.Service
+		err      error
+	)
+
+	s.store.View(func(tx store.ReadTx) {
```

Comment:
Thanks for the stackoverflow link, but I understand the implications. Note that the current code *does* delete secrets that are actually referenced, so by adding this code, we're doing a best-effort attempt at checking conflicts.